### ISSUE TITLE
🐛 Bug Report: a11y - "My org" "Starred" and "Owned" buttons in List Page does not receive special styling upon keyboard focus when it is selected

### ISSUE THREAD

**Issue by logonoff** [backstage/backstage#30401]
### 📜 Issue Labels

- [x] Please familiarize yourself with the issue labels used in this project: [LABELS.md](https://github.com/backstage/backstage/blob/master/LABELS.md)

### 🔎 Search Terms

```plain
ListPage
```

### 🗃️ Project Area

Catalog

### 🔗 External Integration

N/A

### 📝 Description & Context

Buttons in the "Personal" section in the filter sidebar do not receive special styling or a focus ring on keyboard focus when it is selected

### 👍 Expected Behavior

A focus ring or another visual indicator appears around the selected item when it receives focus

### 🥾 Reproduction steps

1. Go to https://demo.backstage.io/catalog?filters%5Bkind%5D=component&filters%5Btype%5D=service&filters%5Blifecycles%5D=production&filters%5Blifecycles%5D=experimental&filters%5Buser%5D=owned&limit=20
2. Select the "Owned" filter with your keyboard
3. Observe that there is no visual difference between when the filter is focused vs when it is selected

### Have you read the Code of Conduct?

- [x] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Undecided

**Comment by MihaiTabara** [comment]
Found in today's triaging meeting.
@logonoff thanks for filing this issue! 

Are you open to contributions to the project and if so, how might we help you bootstrapping in it?

**Comment by MihaiTabara** [comment]
Awesome work. third PR on improving a11y I triage today :)
